File: backend/src/main/java/br/edu/ifs/playifs/game/dto/GameDetailsDTO.java
package br.edu.ifs.playifs.game.dto;

import br.edu.ifs.playifs.game.model.Game;
import br.edu.ifs.playifs.game.model.enums.GamePhase;
import br.edu.ifs.playifs.game.model.enums.GameStatus;
import br.edu.ifs.playifs.team.dto.TeamSummaryDTO;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.Instant;

@Data
@NoArgsConstructor
@Schema(description = "Representa os dados completos de um único jogo.")
public class GameDetailsDTO {

    @Schema(description = "ID único do jogo.", example = "1")
    private Long id;

    @Schema(description = "Data e hora do jogo.")
    private Instant dateTime;

    @Schema(description = "Status atual do jogo.", example = "SCHEDULED")
    private GameStatus status;

    @Schema(description = "Fase da competição a que o jogo pertence.", example = "GROUP_STAGE")
    private GamePhase phase;

    @Schema(description = "Detalhes da Equipa A.")
    private TeamSummaryDTO teamA;

    @Schema(description = "Detalhes da Equipa B.")
    private TeamSummaryDTO teamB;

    @Schema(description = "Placar da Equipa A. Nulo se o jogo não estiver finalizado.", example = "3")
    private Integer scoreTeamA;

    @Schema(description = "Placar da Equipa B. Nulo se o jogo não estiver finalizado.", example = "2")
    private Integer scoreTeamB;

    @Schema(description = "Data e hora da criação do registo.", accessMode = Schema.AccessMode.READ_ONLY)
    private Instant createdAt;

    @Schema(description = "Data e hora da última atualização do registo.", accessMode = Schema.AccessMode.READ_ONLY)
    private Instant updatedAt;

    public GameDetailsDTO(Game entity) {
        this.id = entity.getId();
        this.dateTime = entity.getDateTime();
        this.status = entity.getStatus();
        this.phase = entity.getPhase();
        this.teamA = new TeamSummaryDTO(entity.getTeamA());
        this.teamB = new TeamSummaryDTO(entity.getTeamB());
        this.scoreTeamA = entity.getScoreTeamA();
        this.scoreTeamB = entity.getScoreTeamB();
        this.createdAt = entity.getCreatedAt();
        this.updatedAt = entity.getUpdatedAt();
    }
}

File: frontend/lib/data/models/game/game_details_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:playifs_frontend/data/models/team/team_summary_dto.dart';

part 'game_details_dto.freezed.dart';
part 'game_details_dto.g.dart';

@freezed
abstract class GameDetailsDTO with _$GameDetailsDTO {
  @JsonSerializable(explicitToJson: true, fieldRename: FieldRename.none)
  const factory GameDetailsDTO({
    required int id,
    String? dateTime,
    required String status,
    required String phase,
    TeamSummaryDTO? teamA,
    TeamSummaryDTO? teamB,
    int? scoreTeamA,
    int? scoreTeamB,
    String? createdAt,
    String? updatedAt,
  }) = _GameDetailsDTO;

  factory GameDetailsDTO.fromJson(Map<String, dynamic> json) =>
      _$GameDetailsDTOFromJson(json);
}